php/php.api.phpmodule/src/org/netbeans/modules/php/api/validation/ValidationResult.java

```diff
@@ -80,6 +81,20 @@ public boolean hasErrors() {
         return new ArrayList<>(errors);
     }
 
+    /**
+     * Get the first error.
+     *
+     * @return the first error.
+     * @since 2.72
+     */
+    @CheckForNull
+    public Message getFirstError() {
+        if (hasErrors()) {
+            return errors.get(0);
+        }
+        return null;
+    }
```

Comment:
I also thought other implementations:

#### Create Message.EMPTY and return it 
```java
    public Message getFirstError() {
        if (hasErrors()) {
            return errors.get(0);
        }
        return Message.EMPTY;
    }

    public static final class Message {
        public static final Message EMPTY = new Message("", ""); 

```
#### Use Optional
```java
    public Optional<Message> getFirstError() {
        if (hasErrors()) {
            return Optional.of(errors.get(0));
        }
        return Optional.empty();
    }
```
Which is better?

Reply:
I think that `Message.EMPTY` would complicate things because the caller would need to (a) check whether it is an empty message (which is similar for the current check for `null`) or (b) adjust UI not to show the empty error as an error.

Using `Optional` would be better but I would perhaps keep the current implementation, it is more _consistent_ with the rest of the NetBeans codebase (and we have `@CheckForNull`). But up to you, feel free to use it if you want to, I definitely have no objections.

Thanks.

Reply:
Makes sense. I agree with you. I'll keep this. Thanks a lot!